server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
Both `GaugeService` and `CounterService` are trivial interfaces. I do not see how it would be a problem to provide empty default implementations. I also do not see how the argumentation provided in the criticism section of the Wikipedia article would apply here. Maybe you can help me out?